### ISSUE TITLE
Lint results should not be filtered before being passed to formatter

### ISSUE THREAD

**Issue by tterrag1098** [angular-eslint/angular-eslint#2543]
When using a custom formatter such as [`ratchet`](https://github.com/Jmsa/eslint-formatter-ratchet) the formatter expects to receive results from all processed files, even empty results. When these are filtered out, it results in odd behavior as noted in this issue: https://github.com/Jmsa/eslint-formatter-ratchet/issues/10

I tested removing this filtering in my local environment and can confirm it fixes the issue, and thus must be different from how the "vanilla" eslint command behaves. At the least, this should be a configurable behavior.

To reproduce, simply install and use the ratchet formatter, then fix all errors in any file so that it is filtered from the output. Ratchet will not be able to tell there were changes to this file.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/builder`          | `19.4.0` |
| `ESLint`                           | `9.27.0` |
| `node`                             | `22.14.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:


```

- [ ] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by reduckted** [comment]
> I tested removing this filtering in my local environment

Which filtering are you referring to?

**Comment by tterrag1098** [comment]
https://github.com/angular-eslint/angular-eslint/blob/main/packages/builder/src/lint.impl.ts#L137-L148

This converts any results that do not contain errors/warnings to null, which are then removed, which is different from baseline eslint behavior.

**Comment by reduckted** [comment]
If I'm reading that correctly, that filtering only occurs when the `quiet` option is set to true. It has a default value of false.

**Comment by tterrag1098** [comment]
Read closely, if there are no errors *or warnings* it is always filtered. Quiet just controls whether warnings are considered or not.

**Comment by tknerr** [comment]
Bump, also seeing this here. Was wondering why `npx eslint -f junit` reports *all* processed files, but running `npx ng lint --formatter junit` reports only files with issues or warnings

**Comment by JamesHenry** [comment]
Thanks folks, I'm making this breaking change to the builder in v21 to align with ESLint CLI

**Comment by JamesHenry** [comment]
Released as part of https://www.npmjs.com/package/angular-eslint/v/21.0.0
